File: backend/src/main/java/com/lms/model/Author.java
package com.lms.model;

import java.util.List;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "author")
public class Author {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	@NotNull
	@Column(name = "name", length = 100)
	private String name;

	@NotNull
	@Column(name = "surname", length = 100)
	private String surname;

	@Column(name = "email", length = 100, unique = true)
	private String email;

	@Column(name = "description", length = 8000)
	private String description;

	@OneToMany(mappedBy = "author", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
	private List<Book> books;
}

File: frontend/src/app/demo/api/author/Author.ts
import { Book } from "../book/Book";

export class Author {
    public id!: number;
    public code!: string;
    public name!: string;
    public surname!: string;
    public email!: string;
    public description!: string;
    public books?: Book[];
  }

  export class AuthorDto {
    public id?: number;
    public code?: string;
    public name?: string;
    public surname?: string;
    public email?: string;
    public description?: string;
  }
